docs/rfcs/1012-parsec-auth.md

```diff
@@ -0,0 +1,285 @@
+  S ->> Alice: Retrieve the key `enc(SYM_KEY)` related to `PUB_AUTH_KEY`
+  Alice ->> Alice: Decrypt the key `enc(SYM_KEY)` with `PRIV_AUTH_KEY`
+  Alice ->> Alice: Encrypt the key `SYM_KEY` with `PUB_AUTH_KEY_B` (`enc'(SYM_KEY)`)
+  Alice ->> S: Upload `enc'(SYM_KEY)`, `PUB_AUTH_KEY_B` and the secret of the authentification method
+```
+
+### Removing an authentification method
+
+To remove an authentification method from a user, it require the user to:
+
+1. Retrieve all `PUB_AUTH_KEY` except the one to remove.
+2. Retrieve `enc(SYM_KEY)` related to the current authentification method.
+3. Retrieve all devices `ENC_DEV`.
+4. Decrypt `enc(SYM_KEY)` with `PRIV_AUTH_KEY`.
+5. Create a new symmetric key `SYM_KEY_B`.
+6. For each devices:
+
+   1. Decrypt the device with `SYM_KEY`.
+   2. Encrypt the device with `SYM_KEY_B`.
+
+7. For each `PUB_AUTH_KEY`, encrypt `SYM_KEY` with it.
+8. Upload new list of encrypted devices, the list of `enc(SYM_KEY_B)` of each `PUB_AUTH_KEY`
+9. Remove the authentification method from the server.
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Alice
+  participant S as Authentification server
+
+  Note over Alice,S: Alice is already authenticated to the server
+
+  S ->> Alice: Retrieve every key `PUB_AUTH_KEY` except the one to remove
+  S ->> Alice: Get the symmetric key `enc(SYM_KEY)`
+  S ->> Alice: Download every devices `ENC_DEV`
+  Alice ->> Alice: Decrypt `enc(SYM_KEY)` with `PRIV_AUTH_KEY`
+  Alice ->> Alice: Create a new symmetric key (`SYM_KEY_B`)
+  loop For each device
+    Alice ->> Alice: Decrypt the device with `SYM_KEY`
+    Alice ->> Alice: Encrypt the device with `SYM_KEY_B` (`ENC_DEV_B`)
+  end
+  loop For each `PUB_AUTH_KEY`
+    Alice ->> Alice: Encrypt the key `SYM_KEY_B` with `PUB_AUTH_KEY` (`enc(SYM_KEY_B)`)
+  end
+  Alice ->> S: Upload the list of encrypted device `ENC_DEV_B`
+  Alice ->> S: Upload the list of key `enc(SYM_KEY_B)`
+  Alice ->> S: Remove the authentication method from the server
+```
+
+### Using the service alongside the parsec-cloud server
+
+The parsec client would use the authentification service to store a special device, who will only be use to create a new local device.
+
+That new device would be stored on the local storage and if the storage happen to be cleaned, the user could still re-create a new device from the special one stored on the service.
+
+For already existing organisations and devices, the client could device to create and upload a special device to the service (cf [Uploading a new device](#uploading-a-new-device)).
+
+## Alternatives Considered
+
+N/A
+
+## Operations
+
+That require managing a new service that will store the devices of the user.
+
+## Security/Privacy/Compliance
+
+Consideration should be taken about the method used to save the device on the remote server.
+The service would need to identify the user so we will need to have some information about the user (email mostly).
```

Comment:
Should we generate a parsec auth user id  ? That way a user may have the option to change their email.

Reply:
Indeed that would be better

Reply:
done